src/adapter/src/coord/util.rs

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+
+    /// Verify a prepared statement is still valid.
+    pub(crate) fn verify_prepared_statement(
+        &self,
+        session: &mut Session,
+        name: &str,
+    ) -> Result<(), AdapterError> {
+        let ps = match session.get_prepared_statement_unverified(&name) {
+            Some(ps) => ps,
+            None => return Err(AdapterError::UnknownPreparedStatement(name.to_string())),
+        };
+        if let Some(revision) =
+            self.verify_statement_revision(session, ps.sql(), ps.desc(), ps.catalog_revision)?
+        {
+            let ps = session
+                .get_prepared_statement_mut_unverified(name)
+                .expect("known to exist");
+            ps.catalog_revision = revision;
+        }
+
+        Ok(())
+    }
+
+    /// Verify a portal is still valid.
+    pub(crate) fn verify_portal(
+        &self,
+        session: &mut Session,
+        name: &str,
+    ) -> Result<(), AdapterError> {
+        let portal = match session.get_portal_unverified(&name) {
+            Some(portal) => portal,
+            None => return Err(AdapterError::UnknownCursor(name.to_string())),
+        };
+        if let Some(revision) = self.verify_statement_revision(
+            &session,
+            portal.stmt.as_ref(),
+            &portal.desc,
+            portal.catalog_revision,
+        )? {
+            let portal = session
+                .get_portal_unverified_mut(&name)
+                .expect("known to exist");
+            portal.catalog_revision = revision;
+        }
+        Ok(())
+    }
+
+    fn verify_statement_revision(
+        &self,
+        session: &Session,
+        stmt: Option<&Statement<Raw>>,
+        desc: &StatementDesc,
+        catalog_revision: u64,
+    ) -> Result<Option<u64>, AdapterError> {
+        let current_revision = self.catalog.transient_revision();
+        if catalog_revision != current_revision {
+            let current_desc = self.describe(
+                session,
+                stmt.cloned(),
+                desc.param_types.iter().map(|ty| Some(ty.clone())).collect(),
+            )?;
+            if &current_desc != desc {
+                Err(AdapterError::ChangedPlan)
+            } else {
+                Ok(Some(current_revision))
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Handle removing in-progress transaction state regardless of the end action
+    /// of the transaction.
+    pub(crate) async fn clear_transaction(
+        &mut self,
+        session: &mut Session,
+    ) -> TransactionStatus<mz_repr::Timestamp> {
+        let (drop_sinks, txn) = session.clear_transaction();
+        self.drop_sinks(drop_sinks).await;
+
+        // Release this transaction's compaction hold on collections.
+        if let Some(txn_reads) = self.txn_reads.remove(&session.conn_id()) {
+            self.release_read_hold(&txn_reads.read_holds).await;
+        }
+        txn
+    }
```

Comment:
You mentioned this in the PR description, but I'd be +100 on moving these `mz_adapter::coord::message_procing::sql` module or somesuch. It's kind of funny to have such key methods in a module named `util`.

Reply:
I took some stuff out and renamed it to `sql`.